ttcn3/v2/syntax/syntax.go

```diff
@@ -285,6 +285,12 @@ func (t *tree) position(pos int) Position {
 	if pos < 0 {
 		return Position{}
 	}
+	if t.cachedLineBegin <= pos && pos < t.cachedLineEnd {
+		return Position{
+			Line:   t.cacheLine + 1,
+			Column: pos - t.lines[t.cacheLine] + 1,
+		}
+	}
```

Comment:
Cache-handling is distributed over two methods (`tree.position` and `tree.searchLines`). I suggest you put all cache-related code into only one function. This increases [cohesion](https://en.wikipedia.org/wiki/Cohesion_(computer_science)) and make the code easier to reason about.